atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/metrics/SweepOutcomeMetrics.java

```diff
@@ -71,15 +86,24 @@ public void registerOccurrenceOf(SweepOutcome outcome) {
             MetricsManager manager,
             List<SweepOutcome> outcomes,
             Map<String, String> additionalTags,
-            Class<?> forClass) {
+            Class<?> forClass,
+            Optional<MetricPublicationFilter> filter) {
         return Suppliers.memoize(() -> {
             Metrics metrics = ImmutableSweepOutcomeMetrics.Metrics.builder().build();
-            outcomes.forEach(outcome -> manager.registerOrGet(forClass, AtlasDbMetricNames.SWEEP_OUTCOME,
-                    () -> getOutcomeCount(metrics, outcome),
-                    ImmutableMap.<String, String>builder()
-                            .putAll(additionalTags)
-                            .put(AtlasDbMetricNames.TAG_OUTCOME, outcome.name())
-                            .build()));
+            outcomes.forEach(outcome -> {
+                ImmutableMap<String, String> tags = ImmutableMap.<String, String>builder()
+                        .putAll(additionalTags)
+                        .put(AtlasDbMetricNames.TAG_OUTCOME, outcome.name())
+                        .build();
+                filter.ifPresent(presentFilter -> manager.addMetricFilter(
+                        forClass,
+                        AtlasDbMetricNames.SWEEP_OUTCOME,
+                        tags,
+                        filter.get()));
+                manager.registerOrGet(forClass, AtlasDbMetricNames.SWEEP_OUTCOME,
+                        () -> getOutcomeCount(metrics, outcome),
+                        tags);
```

Comment:
can we have tests for this change?

Reply:
Argh! Well spotted, I forgot to push the commit with the tests :(